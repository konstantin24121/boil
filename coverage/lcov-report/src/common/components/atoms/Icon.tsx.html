<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/common/components/atoms/Icon.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\index.html">All files</a> / <a href="index.html">src/common/components/atoms</a> Icon.tsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">90% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>9/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">90% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/10</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44</td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import * as React from 'react';
import styled, { css } from 'react-emotion';
import { icons, EIconNames } from 'icons/IconsManifest';
&nbsp;
export interface IIconProps extends React.HTMLAttributes&lt;SVGElement&gt; {
  name: EIconNames;
  color?: string;
  width?: string;
  height?: string;
  className?: string;
  rotate?: number;
}
&nbsp;
export class Icon extends React.Component&lt;IIconProps, any&gt; {
  public static defaultProps = {
    width: '1em',
    height: 'auto',
    color: 'currentColor',
    rotate: 0,
  };
  public render() {
    const { name, width, height, color, className, rotate } = this.props;
    const SvgIcon = icons[name].default;
    return (
      &lt;IconStyler width={width} height={height} color={color} className={className} rotate={rotate}&gt;
        &lt;SvgIcon width="100%" height="100%" /&gt;
      &lt;/IconStyler&gt;
    );
  }
}
&nbsp;
const IconStyler = styled('span')&lt;Partial&lt;IIconProps&gt;&gt;(
<span class="fstat-no" title="function not covered" >  ({</span> width, height, color, rotate }) =&gt; <span class="cstat-no" title="statement not covered" >css`</span>
    display: inline-block;
    line-height: 0;
    flex-shrink: 0;
    width: ${width};
    height: ${height};
    transform: rotate(${rotate}deg);
    color: ${color};
    fill: ${color};
  `,
);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Nov 13 2018 00:14:30 GMT+0200 (Греция, Турция (зима))
</div>
</div>
<script src="..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\sorter.js"></script>
</body>
</html>
