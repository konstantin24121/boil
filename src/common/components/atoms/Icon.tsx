import * as React from 'react';
import styled from '@emotion/styled';
import { icons, EIconNames } from 'icons/IconsManifest';
import { css } from '@emotion/core';

export interface IIconProps extends React.HTMLAttributes<SVGElement> {
  name: EIconNames;
  color?: string;
  width?: string;
  height?: string;
  className?: string;
  rotate?: number;
}

export class Icon extends React.Component<IIconProps, {}> {
  public static defaultProps = {
    width: '1em',
    height: 'auto',
    color: 'currentColor',
    rotate: 0,
  };
  public render() {
    const { name, width, height, color, className, rotate } = this.props;
    const SvgIcon = icons[name].default;
    return (
      <IconStyler
        width={width}
        height={height}
        color={color}
        className={className}
        rotate={rotate}
      >
        <SvgIcon width="100%" height="100%" />
      </IconStyler>
    );
  }
}

interface IIconStyledProps
  extends Pick<IIconProps, 'width' | 'height' | 'color' | 'rotate'> {}

const IconStyler = styled('span')<IIconStyledProps>(
  ({ width, height, color, rotate }) => css`
    display: inline-block;
    line-height: 0;
    flex-shrink: 0;
    width: ${width};
    height: ${height};
    transform: rotate(${rotate}deg);
    color: ${color};
    fill: ${color};
  `,
);
