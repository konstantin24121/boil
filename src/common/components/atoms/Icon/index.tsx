import * as React from 'react';
import styled, { css } from 'react-emotion';
import { icons, EIconNames } from 'icons/IconsManifest';

export interface IIconProps extends React.HTMLAttributes<SVGElement> {
  name: EIconNames;
  color?: string;
  width?: string;
  height?: string;
  className?: string;
  rotate?: number;
}

export class Icon extends React.Component<IIconProps, any> {
  public static defaultProps = {
    width: '1em',
    height: 'auto',
    color: 'currentColor',
    rotate: 0,
  };
  public render() {
    const { name, width, height, color, className, rotate } = this.props;
    const SvgIcon = icons[name].default;
    return (
      <IconStyler
        width={width}
        height={height}
        color={color}
        className={className}
        rotate={rotate}
      >
        <SvgIcon width="100%" height="100%" />
      </IconStyler>
    );
  }
}

const IconStyler = styled('span')<Partial<IIconProps>>(
  ({ width, height, color, rotate }) => css`
    display: inline-block;
    line-height: 0;
    flex-shrink: 0;
    width: ${width};
    height: ${height};
    transform: rotate(${rotate}deg);
    color: ${color};
    fill: ${color};
  `,
);
