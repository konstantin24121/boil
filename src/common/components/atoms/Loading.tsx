import * as React from 'react';
import { LoadingComponentProps } from 'react-loadable';

interface IProps extends LoadingComponentProps {}

export class Loading extends React.Component<IProps> {
  public render() {
    return <span>Loading...</span>;
  }
}
