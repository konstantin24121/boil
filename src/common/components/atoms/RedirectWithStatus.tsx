import { Route, Redirect, StaticContext } from 'react-router';
import * as React from 'react';

interface IProps {
  from?: string;
  to: string;
  status: number;
}

export const RedirectWithStatus = ({ from, to, status }) => (
  <Route
    render={({ staticContext }) => {
      if (staticContext) {
        staticContext.statusCode = status;
      }
      return <Redirect from={from} to={to} />;
    }}
  />
);
