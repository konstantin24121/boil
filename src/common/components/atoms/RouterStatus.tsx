import * as React from 'react';
import { Route } from 'react-router';

interface IProps {
  status: number;
  children: React.ReactChild;
}

export const RouterStatus = ({ status, children }: IProps) => (
  <Route
    render={({ staticContext }) => {
      if (staticContext) {
        staticContext.statusCode = status;
      }
      return children;
    }}
  />
);
