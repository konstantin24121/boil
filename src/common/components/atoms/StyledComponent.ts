import styled, { css } from 'react-emotion';

interface IStylesComponentProps extends IThemedComponent {
  underlined?: boolean;
}

const StyledComponent = styled('div')<IStylesComponentProps>(
  ({ underlined, theme }) => {
    return css`
      font-size: 18px;
      color: ${theme.color};
      text-decoration: ${underlined ? 'underline' : undefined};
    `;
  },
);

export { StyledComponent };
