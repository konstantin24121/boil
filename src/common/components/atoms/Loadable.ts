import * as ReactLoadable from 'react-loadable';
import { Loading } from 'atoms/Loading';
import { Omit } from 'utils/Omit';

export const LoadableComponent = (
  options: Omit<ReactLoadable.OptionsWithoutRender<{}>, 'loading'>,
) =>
  ReactLoadable({
    loading: Loading,
    delay: 300,
    ...options,
  });
