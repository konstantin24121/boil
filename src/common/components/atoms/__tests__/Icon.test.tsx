import * as React from 'react';
import * as Enzyme from 'enzyme';
import { EIconNames } from 'icons/IconsManifest';
import { Icon } from 'atoms/Icon';

describe('components/atom/Icon', () => {
  it('Should render correct', () => {
    const snap = Enzyme.shallow(<Icon name={EIconNames.infinity} />);

    expect(snap).toMatchSnapshot();
  });
});
