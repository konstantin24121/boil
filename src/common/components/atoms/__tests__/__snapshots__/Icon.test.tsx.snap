// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atom/Icon Should render correct icon 1`] = `
<Styled(span)
  color="currentColor"
  height="auto"
  rotate={0}
  width="1em"
>
  <Component
    height="100%"
    width="100%"
  />
</Styled(span)>
`;
